src/estimators/any_sketch.py

```diff
@@ -262,7 +306,13 @@ def get_indexes(self, x):
     for index_hashes in self.hash_functions:
       combined_index = []
       for idx_spec, hash_func in zip(self.config.index_specs, index_hashes):
-        combined_index.append(idx_spec.distribution.get_index(hash_func(x)))
+        idx = idx_spec.distribution.get_index(hash_func(x))
+        if idx >= len(idx_spec.distribution):
```

Comment:
Instead of this can you ensure that the geometric distribution does not return an out of bounds value?